wgengine/router/router_linux.go

```diff
@@ -430,6 +433,30 @@ func (r *linuxRouter) Set(cfg *Config) error {
 	return multierr.New(errs...)
 }
 
+var warnStatefulFilteringWithDocker = health.NewWarnable()
+
+func (r *linuxRouter) updateStatefulFilteringWithDockerWarning() {
+	// If stateful filtering is disabled, clear the warning.
+	if !r.statefulFiltering {
+		r.health.SetWarnable(warnStatefulFilteringWithDocker, nil)
+		return
+	}
+
+	// Otherwise, check for the presence of a Docker interface and warn if
+	// it's found on the system.
+	//
+	// TODO(andrew-d): do a better job at detecting Docker, e.g. by looking
+	// for it in the $PATH or by checking for the presence of the Docker
+	// socket/daemon/etc.
+	ifstate := r.netMon.InterfaceState()
+	if _, found := ifstate.Interface["docker0"]; found {
+		r.health.SetWarnable(warnStatefulFilteringWithDocker, fmt.Errorf(""+
+			"Stateful filtering is enabled and Docker was detected; this may prevent Docker containers "+
```

Comment:
prevent other docker containers?

Reply:
I think this applies in both the case where Tailscale is running on the host or in a Docker container? Unless I'm misunderstanding what you mean by "other"?

Reply:
you're right, I was just wondering how we could convey that if it is in running in a container, then that container will be able to do whatever but _other_ containers won't. And if its running on `host` then all containers are "other" containers? I'm sure there is better wording possible here 😓

Reply:
You won't see `docker0` inside of a container, only on the host, unless you use `--network=host`.
I think we only want this warning to print for tailscale on the host, so the current wording seems right.

Reply:
Updated with a comment to make that more clear, and a TODO to maybe add more later